frontend/src/lib/email/ForgotPassword.svelte

```diff
@@ -1,13 +1,23 @@
 <script lang="ts">
   import Email from '$lib/email/Email.svelte';
   import t from '$lib/i18n';
+  import { parseSimpleTimespan, type TimespanComponent } from '$lib/util/timespan';
 
   export let name: string;
   export let resetUrl: string;
+  export let lifetime: string;
+
+  let timecount: number;
+  let timetype: TimespanComponent = 'days';
+  let expirationText: `emails.link_expiration_${TimespanComponent}`;
+  $: expirationText = `emails.link_expiration_${timetype}`;
+  $: [timecount, timetype] = parseSimpleTimespan(lifetime);
+  let expirationParam: Record<TimespanComponent, number> = {};
+  $: { expirationParam = {}; expirationParam[timetype] = timecount; }
```

Comment:
can we pull this all out into a helper function that you pass a lifetime into and it returns a translated string? or a store that returns a translated string? Then we can use that in the other emails too, like project invite and verify email address.

Reply:
Absolutely planning to do so if I decide to keep this code. I just feel like it's significantly overengineered, and I should replace it with a solution that just reports time in days or hours. (At what point outside of testing will we ever send emails with a JWT lifetime measured in seconds, or even minutes?) So I haven't yet pulled this out into a reusable function because if I'm going to replace this with a less-complicated solution, it would be wasted work.